src/cleanup.rs

```diff
@@ -0,0 +1,168 @@
+        cli.extend(
+            [
+                "docker",
+                "run",
+                "--interactive",
+                "--rm",
+                "--network",
+                "host",
+                "-v",
+                &format!("{}:/.kube/config", kube_config),
+                // The empty applyset must be mounted to be seen by the container.
+                "-v",
+                &format!("{}:{}", deletion_dir, deletion_dir),
+                "--env",
+                KUBECTL_APPLYSET_ENABLED,
+                "--env",
+                "KUBECONFIG=/.kube/config",
+                KUBECTL_IMAGE,
+            ]
+            .iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<_>>(),
+        );
+    } else {
+        cli.extend(
+            ["kubectl"]
+                .iter()
+                .map(|s| s.to_string())
+                .collect::<Vec<_>>(),
+        );
+    }
+
+    cli.extend(
+        [
+            "apply",
+            "-n",
+            &app_instance.namespace_any(),
+            "--server-side",
+            "--prune",
+            "--applyset",
+            &app_instance.name_any(),
+            "--field-manager",
+            KUBIT_APPLIER_FIELD_MANAGER,
+            "--force-conflicts",
+            "-v=2",
+            "-f",
+            deletion_dir,
+        ]
+        .iter()
+        .map(|s| s.to_string())
+        .collect::<Vec<_>>(),
+    );
+
+    cli
+}
+
+pub fn emit_post_deletion_commandline(app_instance: &AppInstance, docker: bool) -> Vec<String> {
+    let mut cli: Vec<String> = vec![];
+
+    if docker {
+        let user_home = home_dir().expect("unable to retrieve home directory");
+        let kube_config =
+            env::var("KUBECONFIG").unwrap_or(format!("{}/.kube/config", user_home.display()));
+        cli.extend(
+            [
+                "docker",
+                "run",
+                "--interactive",
+                "--rm",
+                "--network",
+                "host",
+                "-v",
+                &format!("{}:/.kube/config", kube_config),
+                "--env",
+                "KUBECONFIG=/.kube/config",
+                KUBECTL_IMAGE,
+            ]
+            .iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<_>>(),
+        );
+    } else {
+        cli.extend(
+            ["kubectl"]
+                .iter()
+                .map(|s| s.to_string())
+                .collect::<Vec<_>>(),
+        );
+    }
+
+    cli.extend(
+        [
+            "delete",
+            "configmap",
+            &configmap_name_for(app_instance.name_any()),
+            "--namespace",
+            &app_instance.namespace_any(),
+        ]
+        .iter()
+        .map(|s| s.to_string())
+        .collect::<Vec<_>>(),
+    );
+
+    cli
+}
+
+pub fn emit_deletion_setup(ns: &str, output_file: &str) -> Vec<String> {
```

Comment:
I don't think we need to add a special kubit helper script just to produce a configmap yaml.

```bash
kubectl create configmap foo-cleanup --dry-run=client -o yaml
```

By not requiring the kubit image spawned by the kubit controller to be updated right now just to get this new helper command, the development and upgrade process of kubit itself is simplified.

Also another detail: the current `helper cleanup` command generates the configmap name using the `<namespace>-cleanup` name; but there can be multiple appinstances in the same namespace. The name of the configmap should be `<appinstancename>-cleanup`

Reply:
That makes sense :+1: 

My intention here was to keep things in line with the `emit_*` style of how things have been done prior. I've probably abstracted it _too much_ here though :smile: 

I'll rejig this to make it simpler :100: